internal/handler/userHandler_test.go

```diff
@@ -0,0 +1,290 @@
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	user, _ := mapper.UserRequestToUser(requestBody)
+	mockService.EXPECT().SaveUser(ctx, user).Return(nil, errors.New("some error")).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+	assert.Equal(t, http.StatusInternalServerError, recorder.Code)
+	assert.Equal(t, "Something went wrong", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.Nil(t, responseBody.Field)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+func TestBindJsonFailure(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "",
+		LastName:  "",
+		Email:     "",
+		Age:       17,
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code)
+	assert.Equal(t, "Error while binding JSON", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.NotEmpty(t, responseBody.Field)
+
+	fields, _ := convertToMap(responseBody.Field)
+	assert.Equal(t, "required", fields["firstName"])
+	assert.Equal(t, "required", fields["lastName"])
+	assert.Equal(t, "required", fields["email"])
+	assert.Equal(t, "age must be greater than 18", fields["age"])
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+
+func TestBindAgeAndEmailJsonFields(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "John",
+		LastName:  "Doe",
+		Email:     "invalid-email",
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code)
+	assert.Equal(t, "Error while binding JSON", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.NotEmpty(t, responseBody.Field)
+
+	fields, _ := convertToMap(responseBody.Field)
+	assert.Equal(t, "not a valid email address", fields["email"])
+	assert.Equal(t, "required", fields["age"])
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
```

Comment:
These lines are not required, see comment in TestBindJsonFailure

Reply:
Fixed